bottlecap/src/config/mod.rs

```diff
@@ -27,47 +44,22 @@ pub struct Config {
     pub version: Option<String>,
     pub tags: Option<String>,
     pub log_level: LogLevel,
-    pub serverless_logs_enabled: bool,
     #[serde(deserialize_with = "deserialize_processing_rules")]
     pub logs_config_processing_rules: Option<Vec<ProcessingRule>>,
-    pub apm_enabled: bool,
-    pub apm_replace_tags: Option<ObjectIgnore>,
-    pub lambda_handler: String,
     pub serverless_flush_strategy: FlushStrategy,
-    pub trace_enabled: bool,
-    pub serverless_trace_enabled: bool,
-    pub capture_lambda_payload: bool,
-    // ALL ENV VARS below are deprecated or not used by the extension
-    // just here so we don't failover
-    pub flush_to_log: bool,
-    pub logs_injection: bool,
-    pub merge_xray_traces: bool,
-    pub serverless_appsec_enabled: bool,
-    pub extension_version: Option<String>,
     pub enhanced_metrics: bool,
-    pub cold_start_tracing: bool,
-    pub min_cold_start_duration: String,
-    pub cold_start_trace_skip_lib: String,
-    pub service_mapping: Option<String>,
-    pub data_streams_enabled: bool,
-    pub trace_disabled_plugins: Option<String>,
-    pub trace_debug: bool,
+    // Failover
+    pub extension_version: Option<String>,
+    pub serverless_appsec_enabled: bool,
+    pub appsec_enabled: bool,
     pub profiling_enabled: bool,
-    pub git_commit_sha: Option<String>,
-    pub logs_enabled: bool,
-    pub app_key: Option<String>,
-    pub trace_sample_rate: Option<f64>,
-    pub dotnet_tracer_home: Option<String>,
-    pub trace_managed_services: bool,
-    pub runtime_metrics_enabled: bool,
-    pub git_repository_url: Option<String>,
 }
 
 impl Default for Config {
     fn default() -> Self {
         Config {
             // General
-            site: "datadoghq.com".to_string(),
+            site: String::default(),
```

Comment:
Changed it because if you had in `datadog.yaml` something like `datadoghq.eu`, this would be overwritten by the default value, this is wrong. Instead, I just set it after if it's empty.